File: VB/WinSolution.Module.Win/MyWinNewObjectViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Win.SystemModule

Namespace WinSolution.Module.Win
	Public Class MyWinNewObjectViewController
		Inherits WinNewObjectViewController
		Protected Overrides Sub OnAfterConstruction()
			MyBase.OnAfterConstruction()
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			If View.ObjectTypeInfo.Type Is GetType(DomainObject1) Then
				Me.NewObjectAction.Caption = "Add"
			End If
		End Sub
		Protected Overrides Sub OnDeactivating()
			MyBase.OnDeactivating()
			If View.ObjectTypeInfo.Type Is GetType(DomainObject1) Then
				Me.NewObjectAction.Caption = "New"
			End If
		End Sub
	End Class
End Namespace

File: CS/WinSolution.Module.Win/MyWinNewObjectViewController.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Win.SystemModule;

namespace WinSolution.Module.Win {
    public class MyWinNewObjectViewController : WinNewObjectViewController {
        protected override void OnAfterConstruction() {
            base.OnAfterConstruction();
        }
        protected override void OnActivated() {
            base.OnActivated();
            if (View.ObjectTypeInfo.Type==typeof(DomainObject1)){
                this.NewObjectAction.Caption = "Add";                
            }
        }
        protected override void OnDeactivated() {
            base.OnDeactivated();
            if (View.ObjectTypeInfo.Type == typeof(DomainObject1)) {
                this.NewObjectAction.Caption = "New";
            }
        }
    }
}
